scripts/exglobal_stage_ic.py

```diff
@@ -12,36 +12,24 @@
 @logit(logger)
 def main():
 
-    config = cast_strdict_as_dtypedict(os.environ)
-
     # Instantiate the Stage object
-    stage = Stage(config)
-
-    # Pull out all the configuration keys needed to run stage job
-    keys = ['RUN', 'MODE', 'EXP_WARM_START', 'NMEM_ENS',
-            'assim_freq', 'current_cycle', 'previous_cycle',
-            'ROTDIR', 'ICSDIR', 'STAGE_IC_YAML_TMPL', 'DO_JEDIATMVAR',
-            'OCNRES', 'waveGRD', 'ntiles', 'DOIAU',
-            'DO_JEDIOCNVAR', 'DO_STARTMEM_FROM_JEDIICE',
-            'DO_WAVE', 'DO_OCN', 'DO_ICE', 'DO_NEST', 'DO_CA', 'DO_AERO_ANL',
-            'USE_ATM_ENS_PERTURB_FILES', 'USE_OCN_ENS_PERTURB_FILES', 'DO_GSISOILDA', 'DO_LAND_IAU']
-
-    stage_dict = AttrDict()
-    for key in keys:
-        # Make sure OCNRES is three digits
-        if key == "OCNRES":
-            stage.task_config.OCNRES = f"{stage.task_config.OCNRES :03d}"
-        stage_dict[key] = stage.task_config[key]
-
-    # Also import all COM* directory and template variables
-    for key in stage.task_config.keys():
-        if key.startswith("COM"):
-            stage_dict[key] = stage.task_config[key]
-        if "ENSMEM" in stage.task_config:
-            stage_dict["ENSMEM"] = stage.task_config["ENSMEM"]
-
-    # Stage ICs
-    stage.execute_stage(stage_dict)
+    stage = Stage(cast_strdict_as_dtypedict(os.environ))
+
+    # Create staging dictionary with all necessary variables
+    stage_dict = stage.create_stage_dict()
+
+    # Determine member range
+    first_mem, last_mem = stage._select_first_and_final_member(stage_dict.RUN, stage_dict.NMEM_ENS)
+
+    # Loop through members and stage ICs for each
+    for member in range(first_mem, last_mem + 1):
```

Comment:
If a list of members is received from above, this line can become:
```python
for member in member_list:
```
where `member_list = [1, 2, 3, ... ]` or as appropriate